README.md

```diff
@@ -12,10 +12,22 @@
 Finds broken hyperlinks and mail addresses inside Markdown, HTML,
 reStructuredText, or any other text file or website!
 
+<br><hr>
+[🔧  Installation](#installation)
+<br>
+[🎯  Features](#features)
+<br>
+[💻  Usage](#cliusage)
+<br>
+[📚   Library usage](#lib-usage)
+<br>
+[👥   Users](#users)
+<br>
+<br>
```

Comment:
I wonder how we can keep this list in sync if we move around stuff. Isn't there a tool for auto-generating the table of contents?

Reply:
Yes, some tools can automatically update the Table of Contents based on the headings in your document, even if you move or modify the content. One popular tool for generating Table of Content in Markdown is called "markdown-toc." https://github.com/jonschlinkert/markdown-toc

Reply:
Yes sir, I will  explore the implementation details and ensure it aligns seamlessly with the project's goals and guidelines. If it aligns with the idea you have in mind, I will proceed with implementing the solution and submit a pull request for your review , is that alright ?

Reply:
Based on the information you have provided I'm guessing you want me to .Build the Rust port
Run the make toc command in  md-toc command provided by the Rust port and generate the updated Table of Contents for the  README.md file. So that, whenever we need to update the Table of Contents, it automatically  regenerates using the Rust port of "markdown-toc".

Reply:
Yes, that's what I meant.

But now that I think about it, an easier way might be to try out https://github.com/technote-space/toc-generator, which is a GitHub Action, that does everything we need. Their documentation explains how it works.

The Rust thing also works, but it prints the table of contents to `stdout`, so we would have to put the output into the file somehow, e.g. with `sed`. Not a big deal, but an extra step. Also, we'd have to keep the table of contents up-to-date manually. The GitHub action above would take care of that automatically.

I leave it up to you to decide whatever you like to try. 😉

Reply:
Thank you for the suggestions I've reviewed the toc-generator GitHub Action you mentioned, and it seems like a great solution for automatically generating and updating the table of contents . I will follow the documentation provided on the repository to set it up and integrate it into the  workflow ASAP🙌.

Reply:
Thanks a lot, Manish.

Reply:
The required changes have been  made could you please verify?